ktor-server/ktor-server-plugins/ktor-server-auth/jvm/src/io/ktor/server/auth/OAuth.kt

```diff
@@ -117,6 +118,40 @@ public sealed class OAuthServerSettings(public val name: String, public val vers
             emptyList(),
             accessTokenInterceptor
         )
+
+        @Deprecated("This constructor will be removed", level = DeprecationLevel.HIDDEN)
```

Comment:
minor: "This constructor will be removed" -> "Binary compatibility with 2.x"?

Reply:
IMO, deprecation messages should be for users explaining why this is deprecated, not for us explaining why it is kept.

Reply:
AFAIK, declarations with `level=HIDDEN` will be not visible to user, until they open source code. They even should feel this change at all, as new parameter has default value, but ok.

Reply:
Yes, but for consistency, I prefer it this way.